### PR TITLE
[receiver/couchdb] Define metrics

### PR DESCRIPTION
**Pull request by JonathanWamsley**
**Description:** <Describe what has changed.>
This PR adds the metadata for a Couchdb metric receiver.

### CODE REVIEW

File: receiver/couchdbreceiver/metadata.yaml [hunk 1]
```diff
@@ -1 +1,72 @@
...
+  couchdb.httpd.requests:
+    description: The number of HTTP requests by method.
+    unit: "{requests}"
+    sum:
+      monotonic: true
+      aggregation: cumulative
+    attributes: [ node_name, http.method ]
+  couchdb.httpd.responses:
+    description: The number of each HTTP status code.
+    unit: 1
```
Comment (djaglowski) [inline receiver/couchdbreceiver/metadata.yaml:40]:
```suggestion
    unit: "{responses}"
```

**Review by dmitryax — COMMENTED** [review]
@JonathanWamsley can you please rebase? We now require `value_type` to be set for each metrics. 

Also do you mind using the new interface for metric scrapes? It's just adding `--experimental-gen`  flag to go:generate mdatagen command in `codegen.go`. See this PR as example https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6831.

Let me know if you have a chance to try this and please provide feedback if there are any issues

File: receiver/couchdbreceiver/metadata.yaml [hunk 2]
```diff
@@ -1 +1,72 @@
 name: couchdbreceiver
+
+attributes:
+  node_name:
+    description: The name of the node.
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/metadata.yaml]: Changed to couchdb.node.name and removed from listed attributes  under each metric

File: receiver/couchdbreceiver/metadata.yaml [hunk 3]
```diff
@@ -1 +1,72 @@
...
+  http.method:
+    description: An HTTP request method.
+    enum: [ COPY, DELETE, GET, HEAD, OPTIONS, POST, PUT ]
+  http.status_code:
+    description: An HTTP status code.
+    enum: [ http_200, http_201, http_202, http_204, http_206, http_301, http_302, http_304, http_400, http_401, http_403, http_404, http_405, http_406, http_409, http_412, http_413, http_414, http_415, http_416, http_417, http_500, http_501, http_503 ]
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/metadata.yaml]: removed enum line for `http.method` and `http.status_code`

File: receiver/couchdbreceiver/metadata.yaml [hunk 4]
```diff
@@ -1 +1,72 @@
 name: couchdbreceiver
+
+attributes:
+  node_name:
+    description: The name of the node.
+  http.method:
+    description: An HTTP request method.
+    enum: [ COPY, DELETE, GET, HEAD, OPTIONS, POST, PUT ]
+  http.status_code:
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/metadata.yaml]: awesome!

File: receiver/couchdbreceiver/metadata.yaml [hunk 5]
```diff
@@ -1 +1,72 @@
...
+      monotonic: false
+      aggregation: cumulative
+    attributes: [ node_name ]
+  couchdb.database.operations:
+    description: The number of database operations.
+    unit: "{operations}"
+    sum:
+      monotonic: true
+      aggregation: cumulative
+    attributes: [ node_name, operations]
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/metadata.yaml]: `operations` changeds to `operation`

File: receiver/couchdbreceiver/metadata.yaml (hunk 1 above)
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/metadata.yaml]: fixed

File: receiver/couchdbreceiver/metadata.yaml [hunk 6]
```diff
@@ -1 +1,80 @@
...
+  http.status_code:
+    description: An HTTP status code.
+  operation:
+    description: The operation type.
+    enum: [ writes, reads ]
+
+metrics:
+  couchdb.average_request_time:
+    enabled: true
+    description: The length of an average request.
```
Comment (djaglowski) [inline receiver/couchdbreceiver/metadata.yaml:18]:
```suggestion
    description: The average duration of a served request.
```

**Review by djaglowski — COMMENTED** [review]
A couple nits, but looks good.

File: receiver/couchdbreceiver/codegen.go [hunk 7]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 The OpenTelemetry Authors
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/codegen.go]: removed date

File: receiver/couchdbreceiver/metadata.yaml (hunk 6 above)
Comment (JonathanWamsley) [reply receiver/couchdbreceiver/metadata.yaml]: updated description

**Comment by djaglowski** [comment]
Please resolve the conflict

File: versions.yaml [hunk 8]
```diff
@@ -134,13 +134,13 @@ module-sets:
       - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver
       - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/elasticsearchreceiver
       - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver
-      - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fluentforwardreceiver 
-      - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudpubsubreceiver 
-      - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver 
-      - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver 
+      - github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fluentforwardreceiver
```
Comment (dmitryax) [inline versions.yaml:137]: Removing these trailing spaces makes sense, but they are not related to the PR. Could you please remove them? Feel free to submit another PR if you want to

Comment (JonathanWamsley) [reply versions.yaml]: removed

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@JonathanWamsley please rebase and rerun `make generate`. There is some change to the new metrics builder just got merged.

**Review by djaglowski — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**